File: leetsync/3372-longest-strictly-increasing-or-strictly-decreasing-subarray/longest-strictly-increasing-or-strictly-decreasing-subarray.go
func longestMonotonicSubarray(nums []int) int {
	n := len(nums)
	currAsc, currDesc, maxLen := 1, 1, 1
	for i := 0; i < n-1; i++ {
		if nums[i] < nums[i+1] {
			currAsc++
			currDesc = 1
		} else if nums[i] > nums[i+1] {
			currAsc = 1
			currDesc++
		} else {
			currAsc = 1
			currDesc = 1
		}
		maxLen = maxInt(maxLen, maxInt(currAsc, currDesc))
	}
	return maxLen
}
func maxInt(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: leetsync/3372-longest-strictly-increasing-or-strictly-decreasing-subarray/longest-strictly-increasing-or-strictly-decreasing-subarray.java
class Solution {
    public int longestMonotonicSubarray(int[] nums) {
        int n = nums.length;
        int currAsc = 1;
        int currDesc = 1;
        int maxLength = 1;
        for (int i = 0; i < n - 1; i++) {
            if (nums[i] < nums[i + 1]) {
                currAsc++;
                currDesc = 1;
            } else if (nums[i] > nums[i + 1]) {
                currDesc++;
                currAsc = 1;
            } else {
                currAsc = 1;
                currDesc = 1;
            }
            maxLength = Math.max(maxLength, Math.max(currAsc, currDesc));
        }
        return maxLength;
    }
}